core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -379,13 +378,13 @@ final class ShuffleBlockFetcherIterator(
         hostLocalBlocks ++= blocksForAddress.map(info => (info._1, info._3))
         hostLocalBlockBytes += mergedBlockInfos.map(_.size).sum
       } else {
-        remoteBlockBytes += blockInfos.map(_._2).sum
         val (_, timeCost) = Utils.timeTakenMs[Unit] {
           collectFetchRequests(address, blockInfos, collectedRemoteRequests)
         }
         logDebug(s"Collected remote fetch requests for $address in $timeCost ms")
       }
     }
+    val remoteBlockBytes = collectedRemoteRequests.map(_.size).sum
     val numRemoteBlocks = collectedRemoteRequests.map(_.blocks.size).sum
```

Comment:
Instead of two iterations, what about the single iteration? Something like the following?
```scala
val (remoteBlockBytes, numRemoteBlocks) =
    collectedRemoteRequests.reduce((x, y) => (x.size + y.size, x.blocks.size + y.blocks.size))
```